### PR TITLE
Create a contrib module to integrate with django

### PR DESCRIPTION
**Pull request by ebertti**
To install on your django project, you need to register all bots on settings, add one line on
your urls and create a register method that will received a dispatcher to add all handlers of
your bot.

**settings.py**

``` python
TELEGRAM_BOT = [{
    'token': '{YOUR_BOT_TOKEN}',
    'register': 'myapp.dispatcher.register', // register method that will be called
    'webhook': 'https://mydomain.com/telegram_webhook/%s' // optional if you use django.sites
}]
```

**urls.py**

``` python
urlpatterns = [
    ...
    url(r'^telegram_webhook/', include('telegram.contrib.django',)),
    ...
]
```

**myapp/dispatcher.py**

``` python
from telegram.ext import CommandHandler

def start(bot, update):
    bot.sendMessage(update.message.chat_id, text='Hi from telegram bot inside django project!')

def help(bot, update):
    bot.sendMessage(update.message.chat_id, text='This is a help message')

def error(bot, update):
    bot.sendMessage(update.message.chat_id, text='Command not founded.')

// this method will be called on start of application
def register(dispatcher):
    dispatcher.add_handler(CommandHandler("start", start))
    dispatcher.add_handler(CommandHandler("help", help))
    dp.add_error_handler(error)
```

### CODE REVIEW

**Comment by tapionx** [comment]
I like it!

File: telegram/contrib/django.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+    for bot_config in settings.TELEGRAM_BOT:
+        bot = Bot(bot_config['token'])
+
+        if 'webhook' in bot_config:
+            url = bot_config['webhook'] % bot.token
+            if url[-1] != '/':
+                url += '/'
+        else:
+            webhook = reverse('telegram_webhook', kwargs={'token': bot.token})
```
Comment (amywoodehy) [inline telegram/contrib/django.py]: reverse causes circular import. Use instead reverse_lazy()

Comment (amywoodehy) [inline telegram/contrib/django.py]: but in this case we get proxy object. and if we try to evaluate it, we still get circular import error

File: telegram/contrib/django.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            from django.contrib.sites.models import Site
+            current_site = Site.objects.get_current()
+            url = 'https://' + current_site.domain + webhook
+
+        bot.set_webhook(url)
+        bot = Bot(bot_config['token'])
```
Comment (amywoodehy) [inline telegram/contrib/django.py]: why are we creating new Bot object?

File: telegram/contrib/django.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        return HttpResponse()
+
+    def get_dispatcher(self, token):
+        dispatcher = None
+        if token in BOTS_REGISTERED:
+            return BOTS_REGISTERED[token]
+        return dispatcher
+
+
+class DjangoDispatcher(Dispatcher):
```
Comment (amywoodehy) [inline telegram/contrib/django.py]: There is no point of creating custom dispatcher, just use the default one with update_queue=None, workers=0

Comment (ebertti) [inline telegram/contrib/django.py]: Hi @amywoodehy, this is a old PR, based in a old version 😄

Comment (amywoodehy) [inline telegram/contrib/django.py]: @ebertti shall I patch your code or create new fork/pull request?

**Comment by amywoodehy** [comment]
Guys, what if I create django application, that can be stored in contrib folder?

**Comment by tsnoam** [comment]
@amywoodehy That would be highly appreciated. If you have questions or would like to discuss the design, fell free to contact @jh0ker  or myself ( @tsnoam ) on Telegram.

File: telegram/contrib/django.py (hunk 3 above)
Comment (ebertti) [inline telegram/contrib/django.py]: will be great

**Comment by Eldinnie** [comment]
I think the most logical place for this (where I would look for it) would be the examples folder too.

**Comment by thodnev** [comment]
@amywoodehy I like the idea
In that case, bot functionality would largely rely on DB schema. I haven't used Django's ORM (it's total JOIN crap in my opinion), but I used SQLAlchemy a lot. Feel free to ask any related question

**Comment by amywoodehy** [comment]
@thodnev I don't think that we should provide anything like ORM to this integration. Only webhook registration and dispatching to user declared handler

**Comment by bomjacob** [comment]
Closing due to old age, if OP or @amywoodehy wants to make a PR with updated code based on discussion in this thread go ahead :)

**Pull request closed without merging**